### ISSUE TITLE
Version checks are PEP440 compliant not RPM compliant

### ISSUE THREAD

**Issue by schaefi** [SUSE-Enceladus/obs-img-utils#52]
The version check formerly used through ```pkg_resources.parse_version``` and now refactored to use ```packaging.version.Version``` are both implemented to be python PEP440 compliant. However, the implementation is used to check the version and release information of RPM packages and images. For images we are safe as there is only the standard ```major.minor.patch``` version principal  but for RPM packages this version check can fail. For example the package

* apparmor-parser-2.12.2-lp150.6.14.1

is not PEP440 compliant but is RPM compliant.

I suggest to use rpm for that

```python
import rpm
rpm.labelCompare...
```

**Comment by smarlowucf** [comment]
RPM would be a new package requirement?

**Comment by schaefi** [comment]
> RPM would be a new package requirement?

Yes there is ```python3-rpm``` which provides Python Bindings for RPM

**Comment by schaefi** [comment]
Once this is available you can do things like this

```python
import rpm
print(
    rpm.labelCompare(
        ("epoch", "2.12.2", "lp150.6.14.1"),
        ("epoch", "2.12.2", "lp150.6.14.1")
    )
)
```

It either returns ```0``` or ```-1``` for not equal. I think it's not possible to check for older or newer

**Comment by smarlowucf** [comment]
Yep, granted I'm not sure if the example `lp150.6.14.1` is even valid anymore. It seems Leap uses the standard obs build number now that looks like: `150400.1.71` with no letters.

**Comment by smarlowucf** [comment]
Some non-standard examples I see:

```
1.0.3_git201003141408
11.3.0+git1637
3.2.1.git259
3.1.snap20150325
```

And those are package versions.

**Comment by smarlowucf** [comment]
The rpm package does require system bindings are installed which I think limits obs-img-utils to rpm based distros.

**Comment by smarlowucf** [comment]
> The rpm package does require system bindings are installed which I think limits obs-img-utils to rpm based distros.

python3-rpm does exist for Ubuntu which is the Github runner but the package will only work with the base system python. No other pythons will work.
